userspace/libsinsp/logger.cpp

```diff
@@ -137,20 +191,32 @@ void sinsp_logger::log(string msg, severity sev)
 		return;
 	}
 
+	size_t prefix_len = 0;
+	if(m_flags & sinsp_logger::OT_ENCODE_SEV)
+	{
+		static_assert(ENCODE_LEN+1 < sizeof(m_tbuf), "ENCODE_LEN too big");
+		snprintf(m_tbuf, ENCODE_LEN+1, "%s", encode_severity(sev));
+		prefix_len += ENCODE_LEN;
+	}
 	if((m_flags & sinsp_logger::OT_NOTS) == 0)
 	{
 		struct timeval ts;
 		gettimeofday(&ts, NULL);
 		time_t rawtime = (time_t)ts.tv_sec;
 		struct tm* time_info = gmtime(&rawtime);
-		snprintf(m_tbuf, sizeof(m_tbuf), "%.2d-%.2d %.2d:%.2d:%.2d.%.6d ",
+		snprintf(&m_tbuf[prefix_len], sizeof(m_tbuf), "%.2d-%.2d %.2d:%.2d:%.2d.%.6d ",
```

Comment:
shouldn't this be sizeof(m_tbuf) - prefix_len?

Reply:
Looks like it, @bertocci fyi.